packages/liquid-wallet-provider/lib/LiquidWalletProvider.js

```diff
@@ -0,0 +1,174 @@
+import * as liquid from 'liquidjs-lib'
+import * as slip77 from 'slip77'
+import * as bip32 from 'bip32'
+import { mnemonicToSeed } from 'bip39'
+
+
+import { LiquidAddressTypes } from './utils'
+import { Address, asyncSetImmediate } from '@liquality/utils'
+import { ConfidentialAddress } from './confidentialAddress'
+
+
+const ADDRESS_TYPE_TO_PREFIX = {
+  'legacy': 44,
+  'p2sh-segwit': 49,
+  'bech32': 84,
+  'legacy-confidential': 44,
+  'p2sh-segwit-confidential': 49,
+  'blech32': 84
+}
+
+export default class LiquidWalletProvider {
+
+  constructor(network, mnemonic, addressType = 'blech32', superArgs = []) {
```

Comment:
Is a liquid wallet backwards compatible with regular bitcoin wallets? If so, do you think it would make sense that this class actually extends `BitcoinWalletProvider`?

@harshjv  thoughts?

Reply:
The address encoding differs (ie different prefixes and so on) and we have a new subste of addresses type called "confidential addresses" which are like bitcoin (legacy, wrapped segwit, native segwit) with the difference that a blinding public key is encoded inside as well, make them longer than unconfidential one. (ie. bech32 becomes blech32)

I don't think Confidnetial Transactions will ever merged to Bitcoin, too risky. So I would consider it as a different chain/address type.